tests/util.py

```diff
@@ -304,32 +234,131 @@ def deco(*args2, **kwargs2):
-    return new_func
+    """
+    **DEPRECATED**: use tempdir fixture instead.
+    """
+    return func
 
 
-def sprint(*args):
-    sys.stderr.write(' '.join(map(str, args)) + '\n')
+def raises(exc, func, *args, **kwds):
+    """
+    **DEPRECATED**: use pytest.raises instead.
 
+    Raise AssertionError if ``func(*args, **kwds)`` does not raise *exc*.
+    """
+    with pytest.raises(exc):
+        func(*args, **kwds)
 
-_unicode_literals_re = re.compile(r'u(".*?")|u(\'.*?\')')
 
+def raises_msg(exc, msg, func, *args, **kwds):
+    """
+    **DEPRECATED**: use pytest.raises instead.
 
-def remove_unicode_literals(s):
-    return _unicode_literals_re.sub(lambda x: x.group(1) or x.group(2), s)
+    Raise AssertionError if ``func(*args, **kwds)`` does not raise *exc*,
+    and check if the message contains *msg*.
+    """
+    with pytest.raises(exc) as excinfo:
+        func(*args, **kwds)
+    assert msg in str(excinfo.value)
 
 
-def find_files(root, suffix=None):
-    for dirpath, dirs, files in os.walk(root, followlinks=True):
-        dirpath = path(dirpath)
-        for f in [f for f in files if not suffix or f.endswith(suffix)]:
-            fpath = dirpath / f
-            yield os.path.relpath(fpath, root)
+def assert_true(v1, msg=''):
+    """
+    **DEPRECATED**: use assert instead.
+    """
+    assert v1, msg
 
 
-def strip_escseq(text):
-    return re.sub('\x1b.*?m', '', text)
+def assert_equal(v1, v2, msg=''):
+    """
+    **DEPRECATED**: use assert instead.
+    """
+    assert v1 == v2, msg
+
+
+def assert_in(x, thing, msg=''):
+    """
+    **DEPRECATED**: use assert instead.
+    """
+    if x not in thing:
+        assert False, msg or '%r is not in %r' % (x, thing)
+
+
+def assert_not_in(x, thing, msg=''):
+    """
+    **DEPRECATED**: use assert instead.
+    """
+    if x in thing:
+        assert False, msg or '%r is in %r' % (x, thing)
+
+
+class ListOutput(object):
+    """
+    File-like object that collects written text in a list.
+    """
+    def __init__(self, name):
+        self.name = name
+        self.content = []
+
+    def reset(self):
+        del self.content[:]
+
+    def write(self, text):
+        self.content.append(text)
+
+
+# **DEPRECATED**: use pytest.skip instead.
+SkipTest = pytest.skip.Exception
+
+
+class _DeprecationWrapper(object):
+    def __init__(self, mod, deprecated):
+        self._mod = mod
+        self._deprecated = deprecated
+
+    def __getattr__(self, attr):
+        if attr in self._deprecated:
+            obj, instead = self._deprecated[attr]
+            warnings.warn("tests/util.py::%s is deprecated and will be "
+                          "removed in Sphinx 1.7, please use %s instead."
+                          % (attr, instead),
+                          RemovedInSphinx17Warning, stacklevel=2)
+            return obj
+        return getattr(self._mod, attr)
+
+
+sys.modules[__name__] = _DeprecationWrapper(sys.modules[__name__], dict(  # type: ignore
```

Comment:
I think `test/util.py` is not public library. So we don't need to keep compatibility.
Is this really needed?